fiftyone/core/frame.py

```diff
@@ -677,37 +688,57 @@ def _to_frames_dict(self, include_private=False):
         }
 
     def _save_deletions(self, deferred=False):
-        ops = []
+        now = datetime.utcnow()
+
+        sample_ops = []
+        frame_ops = []
 
         if self._delete_all:
-            if deferred:
-                ops.append(DeleteMany({"_sample_id": self._sample_id}))
-            else:
-                self._frame_collection.delete_many(
-                    {"_sample_id": self._sample_id}
+            sample_ops.append(
+                UpdateOne(
+                    {"_id": self._sample_id},
+                    {"$set": {"last_modified_at": now}},
                 )
+            )
+            frame_ops.append(DeleteMany({"_sample_id": self._sample_id}))
+
+            if not deferred:
+                foo.bulk_write(frame_ops, self._frame_collection)
+                foo.bulk_write(sample_ops, self._sample_collection)
 
+                fos.Sample._reload_doc(
+                    self._sample_collection_name, self._sample.id
+                )
                 Frame._reset_docs(
                     self._frame_collection_name, sample_ids=[self._sample.id]
                 )
 
             self._delete_all = False
             self._delete_frames.clear()
-
-        if self._delete_frames:
-            ops = [
-                DeleteOne(
-                    {
-                        "_sample_id": self._sample_id,
-                        "frame_number": frame_number,
-                    }
+        elif self._delete_frames:
+            sample_ops.append(
```

Comment:
While I agree it makes sense to have the sample last updated at reflect deletion in frames, I'm concerned about performance, particularly for large datasets through the api, when nesting the update inside another operation. Perhaps doing it in 2 separate stages so that it can be handled differently through the api?

Reply:
Not sure what you're pointing out. The two `bulk_write()` methods are separate calls, which means separate API calls in the proxy.